lib/screens/tree_list/tree_list_view_model.dart

```diff
@@ -76,4 +84,17 @@ abstract class TreeListViewModelBase with Store, ViewModel {
     fetch(startRow: treeStore.countTreeList() + 1, nbRows: 20)
         .then((_) => isLoadingTrees = false);
   }
-}
+
+  void connectionChanged(dynamic hasConnection) {
+    hasInternetConnection = hasConnection;
+    fetch();
+  }
+
+  void activateConnectionManager() {
+    connectionStatusManager.activateConnectionManager();
+    //Listen for connection change
+    connectionStatusManager.connectionChangeStream
+        .asBroadcastStream()
+        .listen(connectionChanged);
+  }
+}
```

Comment:
Delete these methods from the view Model.
Already activated in the manager itself